rewrite-core/src/test/java/org/openrewrite/table/SearchResultsTest.java

```diff
@@ -0,0 +1,248 @@
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.table;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
+import static org.openrewrite.test.SourceSpecs.text;
+
+class SearchResultsTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromYaml(
+          //language=yml
+          """
+            type: specs.openrewrite.org/v1beta/recipe
+            name: test.SearchMarkerScraping
+            displayName: Find text and add the results to the global datatable
+            description: Hello world.
+            recipeList:
+              - org.openrewrite.text.Find:
+                  find: hi
+                  caseSensitive: true
+              - org.openrewrite.text.Find:
+                  find: SOME OTHER TEXT
+              - org.openrewrite.text.Find:
+                  find: ([A-Z])\\w+
+                  regex: true
+                  caseSensitive: true
+            """,
+          "test.SearchMarkerScraping"
+        );
+    }
+
+    @DocumentExample
+    @Test
+    void searchMarkersAreDetectedDuringRecipeRun() {
+        rewriteRun(
+          spec -> spec.dataTable(SearchResults.Row.class, rows -> assertThat(rows).extracting(
+              SearchResults.Row::getSourcePath,
+              SearchResults.Row::getResult,
+              SearchResults.Row::getParentRecipe,
+              SearchResults.Row::getRecipe)
+            .containsExactlyInAnyOrder(
+              tuple("matched", "hi", "Find text and add the results to the global datatable", "Find text `hi`"),
+              tuple("nested/matched", "hi", "Find text and add the results to the global datatable", "Find text `hi`"),
+              tuple("matched-inside", "some other text", "Find text and add the results to the global datatable", "Find text `SOME OTHER TEXT`")
+            )),
+          text(
+            "hi",
+            "~~>hi",
+            spec -> spec.path("matched")
+          ),
+          text(
+            "hello",
+            spec -> spec.path("non-matched")
+          ),
+          text(
+            "hi",
+            "~~>hi",
+            spec -> spec.path("nested/matched")
+          ),
+          text(
```

Comment:
This seems to be testing the functionality of `org.openrewrite.text.Find` not the data table entry, so would simplify this unit test by removing this case.